src/Apps/Settings/Routes/SavedSearchAlerts/Components/NewSavedSearchAlertEditForm.tsx

```diff
@@ -0,0 +1,419 @@
+import { CriteriaPills } from "Components/Alert/Components/CriteriaPills"
+import { AlertNameInput } from "Components/Alert/Components/Form/AlertNameInput"
+import { Filters } from "Components/Alert/Components/Steps/Filters"
+import CloseIcon from "@artsy/icons/CloseIcon"
+import { AlertFormikValues } from "Components/Alert/Components/Steps/Details"
+import { AlertProvider } from "Components/Alert/AlertProvider"
+import { isEqual } from "lodash"
+import { FiltersFooter } from "Components/Alert/Components/Steps/StepsFooter/FiltersFooter"
+import { ModalHeader } from "Components/Alert/Components/Modal/ModalHeader"
+import { Modal } from "Components/Alert/Components/Modal/Modal"
+import { DEFAULT_FREQUENCY } from "Components/SavedSearchAlert/constants"
+
+interface NewSavedSearchAlertEditFormQueryRendererProps {
+  editAlertEntity: EditAlertEntity
+  onDeleteClick: () => void
+  onCompleted: () => void
+  onCloseClick: () => void
+}
+
+interface NewSavedSearchAlertEditStepsProps {
+  savedSearch: NonNullable<
+    NewSavedSearchAlertEditFormQuery["response"]["me"]
+  >["savedSearch"]
+  viewer: NewSavedSearchAlertEditForm_viewer$data
+  onDeleteClick: () => void
+  onCompleted: () => void
+  onCloseClick: () => void
+}
+
+interface NewSavedSearchAlertEditFormProps {
+  viewer: NewSavedSearchAlertEditForm_viewer$data
+  onDeleteClick: () => void
+  onCompleted: () => void
+}
+const NewSavedSearchAlertEditSteps: React.FC<NewSavedSearchAlertEditStepsProps> = ({
+  viewer,
+  onDeleteClick,
+  onCompleted,
+  onCloseClick,
+}) => {
+  const { current } = useAlertContext()
+
+  return (
+    <>
+      <Media greaterThanOrEqual="md">
+        {current === "ALERT_DETAILS" && (
+          <Box flex={1} p={4}>
+            <Flex justifyContent="space-between">
+              <ModalHeader />
+              <Clickable onClick={onCloseClick}>
+                <CloseIcon display="flex" />
+              </Clickable>
+            </Flex>
+            <Spacer y={4} />
+            <NewSavedSearchAlertEditForm
+              viewer={viewer}
+              onDeleteClick={onDeleteClick}
+              onCompleted={onCompleted}
+            />
+          </Box>
+        )}
+        {current === "ALERT_FILTERS" && (
+          <Box flex={1} px={2} pt={4}>
+            <ModalHeader />
+            <Filters />
+            <FiltersFooter />
+          </Box>
+        )}
+      </Media>
+
+      <Media lessThan="md">
+        <Modal onClose={onCloseClick}>
+          <Flex mx={2}>
+            {current === "ALERT_DETAILS" && (
+              <NewSavedSearchAlertEditForm
+                viewer={viewer}
+                onDeleteClick={onDeleteClick}
+                onCompleted={onCompleted}
+              />
+            )}
+            {current === "ALERT_FILTERS" && <Filters />}
+          </Flex>
+        </Modal>
+      </Media>
+    </>
+  )
+}
+const NewSavedSearchAlertEditForm: React.FC<NewSavedSearchAlertEditFormProps> = ({
```

Comment:
I agree with your point. However, I prefer to have a different component in this case because the design is changing slightly + the alerts work in general is going well and I think we could just easily get rid of the old component very soon.

Reply:
Ah I get it now. You mean adjusting the `Details` component?

Reply:
Would be perfect to do it now. But agree with Devon - it would take some time and I'm afraid we would need another round of QA. I could create a ticket to review and refactor the implementation after the release. How does it sound to you?